components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -881,9 +880,9 @@ private void addAPI(API api, int tenantId) throws APIManagementException {
         int apiId = apiMgtDAO.addAPI(api, tenantId);
         addLocalScopes(api.getId(), tenantId, api.getUriTemplates());
         addURITemplates(apiId, api, tenantId);
-        Map endpointRegistryEntries = APIUtil.extractEndpointRegistryEntries(api.getEndpointConfig());
-        if (endpointRegistryEntries != null) {
-            apiMgtDAO.addAPIRegistryEntryMappings(apiId, endpointRegistryEntries, null);
+        HashSet<String> endpointRegistryEntries = APIUtil.extractEndpointRegistryEntries(api.getEndpointConfig());
+        if (!endpointRegistryEntries.isEmpty()) {
```

Comment:
change variable naming here?

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/8728/commits/216e08f31573e70a482b6f9e3235b139d06c07bf